Core/Src/main.c

```diff
@@ -1,181 +1,249 @@
+extern TIM_HandleTypeDef htim2;
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+
+/* USER CODE BEGIN PV */
+
+/* USER CODE END PV */
+
+/* Private function prototypes -----------------------------------------------*/
+void SystemClock_Config(void);
+/* USER CODE BEGIN PFP */
+
+/* USER CODE END PFP */
+
+/* Private user code ---------------------------------------------------------*/
+/* USER CODE BEGIN 0 */
+
+/* USER CODE END 0 */
+
+/**
+  * @brief  The application entry point.
+  * @retval int
+  */
+int main(void)
+{
+  /* USER CODE BEGIN 1 */
+  uint8_t spi_transmit_data_buf, spi_receive_data_buf = 0;
+  // Need 10 bits to represent a reading
+  uint16_t spi_reading = 0;
+  const uint32_t adc_read_timeout = 5; // In ms
+
+  // Set PWM to minimum duty-cycle
+  __HAL_TIM_SET_COMPARE(&htim2, TIM_CHANNEL_1, ADC_MIN_COUNTS_DUTY_CYCLE);
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);   // Set CS to HIGH
+
+  /* USER CODE END 1 */
+
+  /* MCU Configuration--------------------------------------------------------*/
+
+  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+  HAL_Init();
+
+  /* USER CODE BEGIN Init */
+
+  /* USER CODE END Init */
+
+  /* Configure the system clock */
+  SystemClock_Config();
+
+  /* USER CODE BEGIN SysInit */
+
+  /* USER CODE END SysInit */
+
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+  MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM2_Init();
+  /* USER CODE BEGIN 2 */
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);  // Start the PWM signal
+
+  /* USER CODE END 2 */
+
+  /* Infinite loop */
+  /* USER CODE BEGIN WHILE */
+  while (1) {
+	  loop_start: HAL_Delay(10);
+	  // Note size is in bits
+
+
+	  // --- Take a reading ---
+	  // Set CS to LOW
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);  
+	  // (1) pipe in 0b0000 0001
+	  spi_transmit_data_buf = 0x01;
+	  HAL_SPI_Transmit(&hspi1, &spi_transmit_data_buf,
+			           sizeof(uint8_t), adc_read_timeout);
+
+	  // (2) pipe in/out 0b1000 0000 / 0b0000 00RR
+	  spi_transmit_data_buf = 0xF0;
+	  HAL_SPI_TransmitReceive(&hspi1, &spi_transmit_data_buf,
+			  	  	  	  	  	  	  &spi_receive_data_buf,
+									  sizeof(uint16_t), adc_read_timeout);
+
+	  // Check we read in data properly
+	  if (spi_receive_data_buf > 3) {
+		  // We read data incorrectly, try to reset
+		  goto loop_start;
+	  }
+
+	  spi_reading ^= spi_receive_data_buf;
+	  spi_reading <<= 8; // Shift in in bits 9 and 8
+	  // (3) Grab remaining 8 bits from SPI
+	  spi_receive_data_buf = 0;
+	  HAL_SPI_Receive(&hspi1, &spi_receive_data_buf,
+			          sizeof(uint8_t), adc_read_timeout);
+
+	  spi_reading ^= spi_receive_data_buf;
+	  // Set CS to HIGH, reading is done
+	  HAL_GPIO_TogglePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);

```

Comment:
This entire section can be condensed into 3 lines (CS low, transmit/read, CS high) if you were to implement the size 3 arrays for both transmit and receive.

Reply:
Also I see you are sending the start bit first, thats good! by why are you sending 0xF0? Based on this chart in the DS you are trying to use channel 3, but we want channel 1.
![image](https://user-images.githubusercontent.com/69175242/213581024-eb681ee5-0c89-470a-98c7-4cfaabc4fba2.png)

Reply:
Whoops, mistake on my end -- I made the silly mistake of thinking 0xF0 = 0b1000 0000. What I really wanted was 0x80 :)

Reply:
Leaving unresolved until response on preview review message is addressed.